src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_LENGTH = 10;
+    private static final int FIGURE_COUNT = 5;
+    private static final double DEFAULT_RADIUS = 10.0;
+    private final ColorSupplier colorSupplier;
+    private Random random;
+
+    public FigureSupplier() {
+        colorSupplier = new ColorSupplier();
+        random = new Random();
+    }
+
+    public double getRandomDimension(Random random,int maxCount) {
+        return random.nextDouble() * maxCount;
```

Comment:
1. we can access random field directly
2. we can access MAX_LENGTH constant directly
3. method not used outside, so it should be private
4. if its private, move it after public methods
```suggestion
    private double getRandomDimension() {
        return random.nextDouble() * MAX_LENGTH;
```